samples/Samples.SqlServer/Program.cs

```diff
@@ -1,60 +1,210 @@
 using System;
+using System.Data;
+using System.Data.SqlClient;
 using System.Linq;
-using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
 
 namespace Samples.SqlServer
 {
-    class Program
+    internal class Program
     {
-        static void Main(string[] args)
+        private const string DropCommandText = "DROP TABLE IF EXISTS Employees; CREATE TABLE Employees (Id int PRIMARY KEY CLUSTERED, Name nvarchar(100));";
+        private const string InsertCommandText = "INSERT INTO Employees (Id, Name) VALUES (@Id, @Name);";
+        private const string SelectOneCommandText = "SELECT Name FROM Employees WHERE Id=@Id;";
+        private const string UpdateCommandText = "UPDATE Employees SET Name=@Name WHERE Id=@Id;";
+        private const string SelectManyCommandText = "SELECT * FROM Employees WHERE Id=@Id;";
+        private const string DeleteCommandText = "DELETE FROM Employees WHERE Id=@Id;";
+
+        private static async Task Main()
         {
-            using (var db = new BloggingContext())
+            var connectionString = GetConnectionString();
+
+            using (var connection = new SqlConnection(connectionString))
             {
-                // create database if missing
-                db.Database.EnsureCreated();
+                Console.WriteLine("Calling synchronous methods:");
+                ExecuteQueries(connection);
 
-                var name = "test";
+                Console.WriteLine();
+                Console.WriteLine("Calling asynchronous methods:");
+                await ExecuteQueriesAsync(connection);
+            }
+        }
 
-                var blog = (from b in db.Blogs where b.Name == name select b).FirstOrDefault();
-                if (blog == null)
-                {
-                    blog = new Blog { Name = name };
-                    db.Blogs.Add(blog);
-                    db.SaveChanges();
-                }
+        private static string GetConnectionString()
+        {
+            return Environment.GetEnvironmentVariable("SQLSERVER_CONNECTION_STRING") ??
+                   @"Server=(localdb)\MSSQLLocalDB;Integrated Security=true;";
+        }
+
+        private static void ExecuteQueries(SqlConnection connection)
```

Comment:
I'd really like to de-duplicate these operations if possible because basically the entire structure of the program is replicated in `Samples.DbCommand`, but I'm not quite sure how to do that. Could we put most of this structure in a helper library and each program could provide the helper library with a proxy to modify and execute the DbCommands?

Reply:
This was bothering me, too. Thanks for pushing me to fix. See 6bbfc49a88418fedd9ecdb823dacec71518a6d9a.